### PR TITLE
[O11y][Nagios_xi] Enable secret for the api_key field, update format version and add dashboard-level filters

### PR DESCRIPTION
**Pull request by harnish-elastic**
## Proposed commit message

- Update format version to `3.0.2`.
- Update kibana version to `8.12.0`.
- Inline "by reference" visualizations.
- Added dashboard level filters.
- Enable secrets for sensitive fields.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates #https://github.com/elastic/obs-infraobs-team/issues/1369
- Closes #https://github.com/elastic/obs-infraobs-team/issues/1391

## Screenshots

<details>
  <summary>Nagios XI</summary>

| Events Before | Events After |
|:-:|:-:|
| nagios_xi  events dashboard - before | nagios xi  events dashboard - after |

| Host Before | Host After|
|:-:|:-:|
| nagios xi  host dashboard - before | nagios xi  host dashboard - after |

| Service Before | Service After|
|:-:|:-:|
| nagios_xi  service dashboard-before | nagios_xi  service dashboard-after |

| Overview Before | Overview After|
|:-:|:-:|
| nagios_xi  overview-before | nagios_xi  overview-after |

</details>

### CODE REVIEW

File: packages/nagios_xi/data_stream/service/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -408,7 +408,7 @@ processors:
     ignore_missing: true
     ignore_failure: true
 - set:
-    field: event.dataset
+    field: event.provider
```
[the remark this answers is not in the document]
Comment (harnish-elastic) [reply packages/nagios_xi/data_stream/service/elasticsearch/ingest_pipeline/default.yml:411]: Please refer to this discussion why we need to chose the event.provider. https://github.com/elastic/obs-infraobs-team/issues/1391#issuecomment-2119869994

File: packages/nagios_xi/data_stream/host/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -135,21 +135,22 @@ processors:
...
-    override: true
-    ignore_failure: true
+    value: '{{{json.address}}}'
+    if: ctx.json?.address != null
+    allow_duplicates: false
 - append:
     field: related.ip
-    value: "{{{host.ip}}}"
-    if: ctx?.host?.ip != null
+    value: '{{{json.address}}}'
```
[the remark this answers is not in the document]
Comment (harnish-elastic) [reply packages/nagios_xi/data_stream/host/elasticsearch/ingest_pipeline/default.yml:151]: Null check is already present in both append processor. `nagios_xi.host.address` field data type is `ip`. Now in the json.address field, we are getting string data which is either `"www.example.com"` or `"0.0.0.0"`. So, if the value is `"www.example.com"` then we are simply renaming field to `nagios_xi.host.address`. But if the value contains `"0.0.0.0"` then we need to convert value from string to ip. Since host.ip and related.ip field should be in array, we need to append value to `host.ip` and same goes for `related.ip` field. Also there are null check already present as if condition. Let me know if you still have any queries!

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 3]
```diff
@@ -1,17 +1,51 @@
 # Nagios XI
 
+## Overview
+
 The Nagios XI integration is used to fetch observability data from [Nagios XI](https://www.nagios.org/documentation/) and ingest it into Elasticsearch.
 
+Use the Nagios XI integration to:
+
+- Collect metrics related to the current load, current users, ping, http, ssh, root partition, swap users, total processes, round-trip time, and packet loss. Additionally, gather logs related to system events.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:9]:
```suggestion
- Collect metrics on current load, users, ping, HTTP, SSH, root partition, swap users, total processes, round-trip time, and packet loss, along with system event logs.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 4]
```diff
@@ -1,17 +1,51 @@
...
+- Create visualizations to monitor, measure and analyze the usage trend and key data, and derive business insights.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:10]:
```suggestion
- Create visualizations to monitor, measure, and analyze usage trends and key data for business insights.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 5]
```diff
@@ -1,17 +1,51 @@
...
+
+Metrics offer insights into the performance and health of user's Nagios XI instance. The `Metric` data stream collected by the Nagios XI integration are host and service. These enable users to monitor and troubleshoot the performance of hosts and services within their Nagios XI environment, covering aspects such as `network round trip time`, `packet loss`, `service load`, `user count`, and various other critical indicators.
+
+Data streams:
+- `events`: This data stream gives Nagios XI system event logs.
+- `host`: This data stream gives Nagios XI Host Round Trip Travel Time (rta) and Packet Loss (pl) metrics.
+- `service `: This dataset gives Nagios XI services current load, current users, ping, http, ssh, root partition, swap users and total processes metrics by default.
+
+Note:
+- Users can monitor and see the log inside the ingested documents for Nagios XI in the `logs-*` index pattern from `Discover`, and for metrics, the index pattern is `metrics-*`.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:27]:
```suggestion
- Users can monitor and view logs from the ingested documents for Nagios XI in the `logs-*` index pattern in `Discover`. For metrics, the index pattern is `metrics-*`.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 6]
```diff
@@ -1,17 +1,51 @@
...
+## Data streams
+
+The Nagios XI integration collects logs and metrics data.
+
+Logs provide insights into operations and events within the Nagios XI environment. The `Log` data stream collected by the Nagios XI integration is events. This allows users to track `system events`, understand their causes, and address issues related to infrastructure monitoring and alert management.
+
+Metrics offer insights into the performance and health of user's Nagios XI instance. The `Metric` data stream collected by the Nagios XI integration are host and service. These enable users to monitor and troubleshoot the performance of hosts and services within their Nagios XI environment, covering aspects such as `network round trip time`, `packet loss`, `service load`, `user count`, and various other critical indicators.
+
+Data streams:
+- `events`: This data stream gives Nagios XI system event logs.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:22]:
```suggestion
- `events`: Provides Nagios XI system event logs.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 7]
```diff
@@ -1,17 +1,51 @@
...
+- `host`: This data stream gives Nagios XI Host Round Trip Travel Time (rta) and Packet Loss (pl) metrics.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:23]:
```suggestion
- `host`: Provides Nagios XI Host Round Trip Travel Time (rta) and Packet Loss (pl) metrics.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 8]
```diff
@@ -1,17 +1,51 @@
...
+- `service `: This dataset gives Nagios XI services current load, current users, ping, http, ssh, root partition, swap users and total processes metrics by default.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:24]:
```suggestion
- `service `: Provides Nagios XI service metrics by default, including current load, current users, ping, HTTP, SSH, root partition, swap users, and total processes.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 9]
```diff
@@ -1,17 +1,51 @@
...
+- Users can monitor and see the log inside the ingested documents for Nagios XI in the `logs-*` index pattern from `Discover`, and for metrics, the index pattern is `metrics-*`.
+
 ## Compatibility
 
 This integration has been tested against `Nagios-XI Version: 5.8.7`
 
-## Requirements
+## Prerequisites
+
+User need Elasticsearch for storing and searching user's data and Kibana for visualizing and managing it. User can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended or self-manage the Elastic Stack on user's own hardware.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:35]:
```suggestion
Users need Elasticsearch for storing and searching their data and Kibana for visualizing and managing it. They can either use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on their own hardware.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 10]
```diff
@@ -1,17 +1,51 @@
...
 
 In order to ingest data from Nagios XI:
-- You must know the host for Nagios XI, add that host while configuring the integration package.
+- User must know the host for Nagios XI, add that host while configuring the integration package.
```
Comment (ritalwar) [inline packages/nagios_xi/_dev/build/docs/README.md:38]:
```suggestion
- Users must know the host for Nagios XI and add that host when configuring the integration package.
```

**Comment by harnish-elastic** [comment]
@ritalwar Thanks for review. I have addressed review comments.

File: packages/nagios_xi/data_stream/host/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (devamanv) [reply packages/nagios_xi/data_stream/host/elasticsearch/ingest_pipeline/default.yml:151]: > if the value is "www.example.com" then we are simply renaming field to nagios_xi.host.address

No, what I meant was exactly this case. If renaming happens, how will the value be taken from `json.address` in the two append processors? Are we okay to a blank IP in case that happens?

[the remark this answers is not in the document]
Comment (harnish-elastic) [reply packages/nagios_xi/data_stream/host/elasticsearch/ingest_pipeline/default.yml:151]: If rename happens, the IP fields will not be generated in this case. For more information, please refer [pipeline tests](https://github.com/elastic/integrations/pull/9996/files#diff-390d4ecb4ab853544594ec61ac7509f545f36e823a743022bf6aac2fb3de2cd4R21).

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 11]
```diff
@@ -1,17 +1,51 @@
...
+## Setup
+
+For step-by-step instructions on how to set up an integration, see the [Getting started](https://www.elastic.co/guide/en/welcome-to-elastic/current/getting-started-observability.html) guide.
+
+## Validation
+
+After the integration is successfully configured, clicking on the Assets tab of the Nagios XI Integration should display a list of available dashboards. Click on the dashboard available for user's configured data stream. It should be populated with the required data.
 
-## Logs
+## Logs reference
```
Comment (devamanv) [inline packages/nagios_xi/_dev/build/docs/README.md:48]: Why this change? Is it in accordance to any guideline, or any other example where we may be doing the same?

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 12]
```diff
@@ -1,17 +1,51 @@
...
+
+Metrics offer insights into the performance and health of user's Nagios XI instance. The `Metric` data stream collected by the Nagios XI integration are host and service. These enable users to monitor and troubleshoot the performance of hosts and services within their Nagios XI environment, covering aspects such as `network round trip time`, `packet loss`, `service load`, `user count`, and various other critical indicators.
+
+Data streams:
+- `events`: Provides Nagios XI system event logs.
+- `host`: Provides Nagios XI Host Round Trip Travel Time (rta) and Packet Loss (pl) metrics.
+- `service `: Provides Nagios XI service metrics by default, including current load, current users, ping, HTTP, SSH, root partition, swap users, and total processes.
+
+Note:
+- Users can monitor and view logs from the ingested documents for Nagios XI in the `logs-*` index pattern in `Discover`. For metrics, the index pattern is `metrics-*`.
```
Comment (devamanv) [inline packages/nagios_xi/_dev/build/docs/README.md:27]:
```suggestion
You can monitor and view logs from the ingested documents for Nagios XI in the `logs-*` index pattern in `Discover`. For metrics, the index pattern is `metrics-*`.
```

File: packages/nagios_xi/_dev/build/docs/README.md [hunk 13]
```diff
@@ -1,17 +1,51 @@
...
+- Create visualizations to monitor, measure, and analyze usage trends and key data for business insights.
+- Set up alerts to minimize Mean Time to Detect (MTTD) and Mean Time to Resolve (MTTR) by quickly referencing relevant logs during troubleshooting.
+
+## Data streams
+
+The Nagios XI integration collects logs and metrics data.
+
+Logs provide insights into operations and events within the Nagios XI environment. The `Log` data stream collected by the Nagios XI integration is events. This allows users to track `system events`, understand their causes, and address issues related to infrastructure monitoring and alert management.
+
+Metrics offer insights into the performance and health of user's Nagios XI instance. The `Metric` data stream collected by the Nagios XI integration are host and service. These enable users to monitor and troubleshoot the performance of hosts and services within their Nagios XI environment, covering aspects such as `network round trip time`, `packet loss`, `service load`, `user count`, and various other critical indicators.
```
Comment (devamanv) [inline packages/nagios_xi/_dev/build/docs/README.md:19]:
```suggestion
Metrics provide insights into the performance and health of your Nagios XI instance. The Nagios XI integration collects `host` and `service` metric data streams. These metrics enable you to monitor and troubleshoot the performance of hosts and services within your Nagios XI environment, covering aspects such as `network round trip time`, `packet loss`, `service load`, `user count`, and other critical indicators. 
```
Change `users` to `yours` for a more direct address to the reader. We follow the same practice for other packages as well.

**Review by devamanv — APPROVED** [review]
LGTM

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Review by ritalwar — APPROVED** [review]
